gdal/ogr/ogrsf_frmts/flatgeobuf/geometryreader.cpp

```diff
@@ -409,9 +408,9 @@ OGRPolyhedralSurface *GeometryReader::readPolyhedralSurface()
 
 OGRTriangulatedSurface *GeometryReader::readTIN()
 {
-    const auto pEnds = m_geometry->ends();
+    const auto ends = m_geometry->ends();
     const auto ts = new OGRTriangulatedSurface();
```

Comment:
By the way, why not to follow this exception safety convention that is already in use here?

https://github.com/OSGeo/gdal/blob/1a539bb944ebaa10d11bd32a8c78113c5dc3b9b4/gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobufdataset.cpp#L185-L186

Reply:
Merging as in current state. I'll let @bjornharrtell issue a potential follow-up PR if he feels like using more unique_ptr goodness